app/models/git_repository.rb

```diff
@@ -1,22 +1,23 @@
+require "rugged"
+
 class GitRepository < ApplicationRecord
   include AuthenticationMixin
 
   GIT_REPO_DIRECTORY = Rails.root.join('data/git_repos')
 
-  validates :url, :format => URI::regexp(%w(http https)), :allow_nil => false
-  validate  :check_path
+  validates :url, :format => URI::regexp(%w(http https file)), :allow_nil => false
 
   default_value_for :verify_ssl, OpenSSL::SSL::VERIFY_PEER
   validates :verify_ssl, :inclusion => {:in => [OpenSSL::SSL::VERIFY_NONE, OpenSSL::SSL::VERIFY_PEER]}
 
   has_many :git_branches, :dependent => :destroy
   has_many :git_tags, :dependent => :destroy
-  after_destroy :delete_repo_dir
+  after_destroy :delete_repo_dir # TODO: Need to distribute this to all systems
```

Comment:
The problem becomes more apparent when we get to multiple embedded_ansible roles allowed.  Then, the repo can live on multiple machines and we need to broadcast the delete.

Reply:
Right, mostly what I was getting at is at most these means we have some orphaned directories sitting on some boxes that nothing will end up using (at least as far as MIQ is concerned), but it shouldn't hurt anything besides disk space.

Again, it is a good idea to clean it up, just don't think it would hurt anything, right?

Reply:
Correct, and I plan to cleanup in follow up PRs.